repo/local_config.go

```diff
@@ -88,26 +90,34 @@ type repositoryObjectFormat struct {
 	object.Format
 }
 
-// Load reads local configuration from the specified reader.
-func (lc *LocalConfig) Load(r io.Reader) error {
-	*lc = LocalConfig{}
-	return json.NewDecoder(r).Decode(lc)
-}
+// writeToFile writes the config to a given file.
+func (lc *LocalConfig) writeToFile(filename string) error {
+	lc2 := *lc
+
+	if lc.Caching != nil {
+		lc2.Caching = lc.Caching.CloneOrDefault()
+
+		// try computing relative pathname from config dir to the cache dir.
+		d, err := filepath.Rel(filepath.Dir(filename), lc.Caching.CacheDirectory)
+		if err == nil {
+			lc2.Caching.CacheDirectory = d
+		}
+	}
 
-// Save writes the configuration to the specified writer.
-func (lc *LocalConfig) Save(w io.Writer) error {
-	b, err := json.MarshalIndent(lc, "", "  ")
+	b, err := json.MarshalIndent(lc2, "", "  ")
 	if err != nil {
-		return nil
+		return errors.Wrap(err, "error writing config file")
 	}
 
-	_, err = w.Write(b)
+	if err = os.MkdirAll(filepath.Dir(filename), 0o700); err != nil {
+		return errors.Wrap(err, "unable to create config directory")
+	}
 
-	return errors.Wrap(err, "error saving local config")
+	return atomicfile.Write(filename, bytes.NewReader(b))
```

Comment:
Should the error be wrapped?
`errors.Wrap` does the right thing when `err == nil`